src/rules/utils.ts

```diff
@@ -648,33 +648,126 @@ export const getTestCallExpressionsFromDeclaredVariables = (
+              isTestCaseCall(node),
           ),
       ),
     [],
   );
 };
 
-export const isTestCase = (
+const isTestCaseName = (node: TSESTree.LeftHandSideExpression) =>
+  node.type === AST_NODE_TYPES.Identifier &&
+  TestCaseName.hasOwnProperty(node.name);
+
+const isTestCaseProperty = (
+  node: TSESTree.Expression,
+): node is AccessorNode<TestCaseProperty> =>
+  isSupportedAccessor(node) &&
+  TestCaseProperty.hasOwnProperty(getAccessorValue(node));
+
+/**
+ * Checks if the given `node` is a *call* to a test case function that would
+ * result in tests being run by `jest`.
+ *
+ * Note that `.each()` does not count as a call in this context, as it will not
+ * result in `jest` running any tests.
+ *
+ * @param {TSESTree.CallExpression} node
+ *
+ * @return {node is JestFunctionCallExpression<TestCaseName>}
+ */
+export const isTestCaseCall = (
   node: TSESTree.CallExpression,
-): node is JestFunctionCallExpression<TestCaseName> =>
-  (node.callee.type === AST_NODE_TYPES.Identifier &&
-    TestCaseName.hasOwnProperty(node.callee.name)) ||
-  // e.g. it.each``()
-  (node.callee.type === AST_NODE_TYPES.TaggedTemplateExpression &&
-    node.callee.tag.type === AST_NODE_TYPES.MemberExpression &&
-    node.callee.tag.object.type === AST_NODE_TYPES.Identifier &&
-    TestCaseName.hasOwnProperty(node.callee.tag.object.name) &&
-    isSupportedAccessor(node.callee.tag.property, TestCaseProperty.each)) ||
-  // e.g. it.concurrent.{skip,only}
-  (node.callee.type === AST_NODE_TYPES.MemberExpression &&
-    node.callee.property.type === AST_NODE_TYPES.Identifier &&
-    TestCaseProperty.hasOwnProperty(node.callee.property.name) &&
-    ((node.callee.object.type === AST_NODE_TYPES.Identifier &&
-      TestCaseName.hasOwnProperty(node.callee.object.name)) ||
-      (node.callee.object.type === AST_NODE_TYPES.MemberExpression &&
-        node.callee.object.object.type === AST_NODE_TYPES.Identifier &&
-        TestCaseName.hasOwnProperty(node.callee.object.object.name))));
+): node is JestFunctionCallExpression<TestCaseName> => {
+  if (isTestCaseName(node.callee)) {
+    return true;
+  }
+
+  if (
+    node.callee.type === AST_NODE_TYPES.MemberExpression &&
+    isTestCaseProperty(node.callee.property)
+  ) {
+    // if we're an `each`, ensure we're being called (i.e `.each()()`)
+    if (
+      getAccessorValue(node.callee.property) === 'each' &&
+      node.parent?.type !== AST_NODE_TYPES.CallExpression
+    ) {
+      return false;
+    }
+
+    return node.callee.object.type === AST_NODE_TYPES.MemberExpression
+      ? isTestCaseName(node.callee.object.object)
+      : isTestCaseName(node.callee.object);
+  }
+
+  if (
+    node.callee.type === AST_NODE_TYPES.TaggedTemplateExpression &&
+    node.callee.tag.type === AST_NODE_TYPES.MemberExpression
+  ) {
+    return node.callee.tag.object.type === AST_NODE_TYPES.MemberExpression
+      ? isTestCaseName(node.callee.tag.object.object)
+      : isTestCaseName(node.callee.tag.object);
```

Comment:
I'm *sure* there's a way to restructure this so we don't have to do it twice, but can't figure it out.

I'm also sure that the same restructure will be usable in `no-focused-tests` (as it has pretty much the exact same conditional structure)